CRM/Contact/BAO/Query.php

```diff
@@ -2623,27 +2624,29 @@ public static function fromClause(&$tables, $inner = NULL, $right = NULL, $prima
         }
         continue;
       }
+      $searchPrimary = '';
+      if (Civi::settings()->get('searchPrimaryLocTypes') || $apiEntity) {
+        $searchPrimary = " AND {$name}.is_primary = 1";
```

Comment:
@jitendrapurohit I think its the space here before the AND that is generating the test fail